crossdock/client/random/rand.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package random
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+	"io/ioutil"
+)
+
+// Bytes generates the gvien number of random bytes.
+func Bytes(length int) []byte {
+	out := make([]byte, length)
+	if _, err := io.ReadFull(rand.Reader, out); err != nil {
+		panic(err)
+	}
+	return out
+}
+
+// String generates a random string of the given size.
+func String(length int64) string {
+	bs, err := ioutil.ReadAll(io.LimitReader(rand.Reader, length))
+	if err != nil {
+		panic(err)
+	}
+	return base64.RawStdEncoding.EncodeToString(bs)
```

Comment:
Better yet: I can just slice the resulting string down to the actual requested size.